diskfs_darwin.go

```diff
@@ -14,13 +14,24 @@ const (
 	DKIOCGETBLOCKCOUNT        = 0x40086419
 )
 
+// getBlockDeviceSize get the size of an opened block device in Bytes.
+func getBlockDeviceSize(f *os.File) (int64, error) {
+	fd := f.Fd()
+
+	blockSize, err := unix.IoctlGetInt(int(fd), DKIOCGETBLOCKSIZE)
+	if err != nil {
+		return 0, fmt.Errorf("unable to get device logical sector size: %v", err)
+	}
+
+	blockCount, err := unix.IoctlGetInt(int(fd), DKIOCGETBLOCKCOUNT)
+	if err != nil {
+		return 0, fmt.Errorf("unable to get device block count: %v", err)
+	}
+	return int64(blockSize) * int64(blockCount), nil
+}
+
 // getSectorSizes get the logical and physical sector sizes for a block device
 func getSectorSizes(f *os.File) (logicalSectorSize, physicalSectorSize int64, err error) {
-	//nolint:gocritic // we keep this for reference to the underlying syscall
-	/*
-		ioctl(fd, BLKPBSZGET, &physicalsectsize);
-
-	*/
```

Comment:
Why remove this comment?

Reply:
Because it looks strange to keep a reference to a Linux ioctl in Darwin code.

Reply:
Fair enough.